contracts/EthStorageContractM1.sol

```diff
@@ -18,10 +18,22 @@ contract EthStorageContractM1 is EthStorageContract, Decoder {
     error EthStorageContractM1_InvalidSamples();
 
     /// @notice Constructs the EthStorageContractM1 contract.
+    /// @custom:oz-upgrades-unsafe-allow constructor
     constructor(Config memory _config, uint256 _startTime, uint256 _storageCost, uint256 _dcfFactor)
         EthStorageContract(_config, _startTime, _storageCost, _dcfFactor)
     {}
 
+    /// @notice Initialize the contract
+    function initialize(
```

Comment:
I’m wondering why an `initialize` function is required in this contract. Could you clarify the rationale?